kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -0,0 +1,193 @@
+        >>>
+        >>> dataset = TableDataset(
+        ...     table_name="test",
+        ...     connection={"backend": "duckdb", "database": tmp_path / "file.db"},
+        ...     save_args={"materialized": "table"},
+        ... )
+        >>> dataset.save(data)
+        >>> reloaded = dataset.load()
+        >>> assert data.execute().equals(reloaded.execute())
+
+    """
+
+    DEFAULT_LOAD_ARGS: ClassVar[dict[str, Any]] = {}
+    DEFAULT_SAVE_ARGS: ClassVar[dict[str, Any]] = {
+        "materialized": "view",
+        "overwrite": True,
+    }
+
+    _connections: ClassVar[dict[tuple[tuple[str, str]], BaseBackend]] = {}
+
+    def __init__(  # noqa: PLR0913
+        self,
+        *,
+        filepath: str | None = None,
+        file_format: str | None = None,
+        table_name: str | None = None,
+        connection: dict[str, Any] | None = None,
+        load_args: dict[str, Any] | None = None,
+        save_args: dict[str, Any] | None = None,
+    ) -> None:
+        """Creates a new ``TableDataset`` pointing to a table (or file).
+
+        ``TableDataset`` connects to the Ibis backend object constructed
+        from the connection configuration. The `backend` key provided in
+        the config can be any of the `supported backends <https://ibis-\
+        project.org/install>`_. The remaining dictionary entries will be
+        passed as arguments to the underlying ``connect()`` method (e.g.
+        `ibis.duckdb.connect() <https://ibis-project.org/backends/duckdb\
+        #ibis.duckdb.connect>`_).
+
+        If ``filepath`` and ``file_format`` are given, the corresponding
+        read method (e.g. `read_csv() <https://ibis-project.org/backends/\
+        duckdb#ibis.backends.duckdb.Backend.read_csv>`_) is used to load
+        the file with the backend. Note that only the data is loaded; no
+        link to the underlying file exists past ``TableDataset.load()``.
+
+        If ``table_name`` is given (and ``filepath`` isn't), the dataset
+        establishes a connection to the relevant table for the execution
+        backend. Therefore, Ibis doesn't fetch data on load; all compute
+        is deferred until materialization, when the expression is saved.
+        In practice, this happens when another ``TableDataset`` instance
+        is saved, after running code defined across one more more nodes.
+
+        Args:
+            filepath: Path to a file to register as a table. Most useful
+                for loading data into your data warehouse (for testing).
+            file_format: Specifies the input file format for `filepath`.
+            table_name: The name of the table or view to read or create.
+            connection: Configuration for connecting to an Ibis backend.
+            load_args: Additional arguments passed to the Ibis backend's
+                `read_{file_format}` method.
+            save_args: Additional arguments passed to the Ibis backend's
```

Comment:
Any docs we can link to that specify which values a user can supply to `materialized`?

Reply:
I'm not aware of anywhere it's documented centrally (i.e. not on a backend-specific basis) in Ibis. @lostmygithubaccount any chance you know?

Reply:
we do not, this is on my TODO list to add API docs for alongside the other `read_*` and `to_*` methods

Reply:
Sounds good! @merelcht if it's OK then, we'll improve the documentation on the Ibis side, and then whenever that happens, we can make the dataset docs reference that.